internal/util/util.go

```diff
@@ -22,3 +31,45 @@ func LookupImage(ctx *types.SystemContext, store storage.Store, image string) (*
 	}
 	return localImage, nil
 }
+
+// ExportFromReader reads bytes from given reader and exports to external tar, directory or stdout.
+func ExportFromReader(input io.Reader, opts define.BuildOutputOption) error {
+	var err error
+	if !filepath.IsAbs(opts.Path) {
+		opts.Path, err = filepath.Abs(opts.Path)
+		if err != nil {
+			return err
+		}
+	}
+	if opts.IsDir {
+		// preserve ownership when invoked as root
```

Comment:
Same documentation point here.

(And if the two are parts of a single idea, as it seems to be the case, it would be nice to at least have comments that link the two. Or perhaps an explicit code link — have a `preserveOwnership bool` parameter to this function, and have the caller set it along with the three set-UID/set-GUID/xattr choices, in a single `if` block.)

Reply:
Added a reason and link to the thread.